sr_utilities_common/src/sr_utilities_common/aws_manager.py

```diff
@@ -147,16 +166,19 @@ def gather_all_files_local(files_base_path, files_folder_path):
 
 
 def validated_files_to_be_downloaded(bucket_name, files_base_path, files_folder_path,
-                                     file_names, bucket_subfolder):
+                                     file_names, bucket_subfolder, preserve_downloaded_folder_structure=False):
     print_msg = f"\nFrom bucket {bucket_name} downloading the files:"
     print_msg_2 = ""
     for filename in file_names:
-        print_msg_2 += f"\n    {files_base_path}/{files_folder_path}/{filename}"
+        if preserve_downloaded_folder_structure and bucket_subfolder:
+            print_msg_2 += f"\n    {files_base_path}/{files_folder_path}/{bucket_subfolder}/{filename}"
+        else:
+            print_msg_2 += f"\n    {files_base_path}/{files_folder_path}/{filename}"
         if bucket_subfolder:
             print_msg += f"\n    {bucket_subfolder}/{filename}"
         else:
             print_msg += f"\n    {filename}"
-    print_msg += f"\n\nThese files will be downloaded too {files_base_path}/{files_folder_path}:"
+    print_msg += f"\n\nThese files will be downloaded to {files_base_path}/{files_folder_path}:"
     print_msg += print_msg_2
     rospy.loginfo(print_msg)
     value = input("Would you like to download these files? (Y/N) ")
```

Comment:
Would suggest lowercase `(y/n)` instead of `(Y/N)`

Reply:
that whole thing should be refactored, but done